src/Stripe.net/Services/_interfaces/ISingletonRetrievable.cs

```diff
@@ -3,6 +3,8 @@ namespace Stripe
     using System.Threading;
     using System.Threading.Tasks;
 
+    // TODO (MAJOR): Accept a TOptions generic and use it to define methods
```

Comment:
I'm not sure whether this change is worth making. We can define another interface later if there is interest.

Reply:
Sounds good, I'll remove this TODO